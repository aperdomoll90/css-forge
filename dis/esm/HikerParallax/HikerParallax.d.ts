import './styles.css';
import '../utils/GlobalStyles.css';
export declare const HikerParallax: React.FC<{}>;
