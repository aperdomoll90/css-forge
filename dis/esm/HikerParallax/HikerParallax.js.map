{"version":3,"file":"HikerParallax.js","sourceRoot":"","sources":["../../../src/HikerParallax/HikerParallax.tsx"],"names":[],"mappings":";;;;;;;;;;;;AAAA,OAAO,EAAE,SAAS,EAAE,MAAM,OAAO,CAAA;AACjC,OAAO,cAAc,CAAA;AACrB,OAAO,2BAA2B,CAAA;AAClC,wCAAwC;AACxC,0CAA0C;AAC1C,0CAA0C;AAC1C,sDAAsD;AACtD,OAAO,GAAG,MAAM,oBAAoB,CAAA;AAEpC,MAAM,CAAC,IAAM,aAAa,GAAiB;IACzC,SAAS,CAAC;QACR,MAAM,CAAC,gBAAgB,CAAC,QAAQ,EAAE;YAChC,IAAI,MAAM,GAAG,MAAM,CAAC,WAAW,CAAA;YAC/B,IAAI,WAAW,GAAG,MAAM,CAAC,MAAM,CAAC,KAAK,CAAA;YAErC,IAAM,SAAS,GAAG,QAAQ,CAAC,gBAAgB,CAAC,YAAY,CAAQ,CAAA;YAChE,IAAM,kBAAkB,GAAG,QAAQ,CAAC,aAAa,CAAC,qBAAqB,CAAQ,CAAA;YAC/E,IAAM,MAAM,GAAG,QAAQ,CAAC,aAAa,CAAC,aAAa,CAAQ,CAAA;YAE3D,kCAAkC;YAClC,SAAS,CAAC,OAAO,CAAC,UAAC,OAAY;gBAC7B,IAAI,KAAK,GAAG,OAAO,CAAC,OAAO,CAAC,KAAK,CAAA;gBACjC,OAAO,CAAC,KAAK,CAAC,SAAS,GAAG,qBAAc,MAAM,GAAG,KAAK,QAAK,CAAA;YAC7D,CAAC,CAAC,CAAA;YAEF,8BAA8B;YAC9B,IAAI,MAAM,EAAE;gBACV,MAAM,CAAC,KAAK,CAAC,MAAM,GAAG,UAAG,MAAM,GAAG,GAAG,GAAG,GAAG,OAAI,CAAA;aAChD;YAED,6BAA6B;YAC7B,IAAI,kBAAkB,EAAE;gBACtB,IAAI,WAAW,GAAG,GAAG,EAAE;oBACrB,kBAAkB,CAAC,KAAK,CAAC,KAAK,GAAG,UAAG,CAAC,MAAM,GAAG,EAAE,GAAG,GAAG,OAAI,CAAA;iBAC3D;gBACD,IAAI,WAAW,GAAG,GAAG,EAAE;oBACrB,kBAAkB,CAAC,KAAK,CAAC,KAAK,GAAG,UAAG,CAAC,MAAM,GAAG,EAAE,GAAG,GAAG,OAAI,CAAA;iBAC3D;aACF;QACH,CAAC,CAAC,CAAA;IACJ,CAAC,EAAE,EAAE,CAAC,CAAA;IACN,OAAO,CACL,wBAAK,EAAE,EAAC,gBAAgB,iBACtB,+BAAW,EAKX,cAAK,GAAG,EAAE,GAAG,EAAE,SAAS,EAAC,eAAe,gBAAY,MAAM,EAAC,GAAG,EAAC,KAAK,GAAG,KACnE,CACP,CAAA;AACH,CAAC,CAAA"}