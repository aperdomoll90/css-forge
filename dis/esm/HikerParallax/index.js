export { HikerParallax } from "./HikerParallax";
//# sourceMappingURL=index.js.map