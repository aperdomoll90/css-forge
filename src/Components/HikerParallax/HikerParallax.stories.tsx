import { Meta, StoryFn } from '@storybook/react'
import HikerParallax from './HikerParallax'
import StorybookContainer from '../StorybookContainer'

export default {
  title: 'Parallax',
  component: HikerParallax,
  parameters: {
    jest: ['HikerParallax.test.tsx'],
  },
} as Meta

const HikerParallaxTemplate: StoryFn = () => (
  <StorybookContainer>
    <div
      style={{
        position: 'relative',
        width: '100vw',
        minHeight: '200vh',
        overflow: 'scroll',
        background:'yellow',
      }}>
      <HikerParallax />
      <div
      style={{
        position: 'relative',
        width: '100vw',
        minHeight: '100vh',
        background:'red',
      }}/>
    </div>
  </StorybookContainer>
)

export const HikerParallaxComponent = HikerParallaxTemplate.bind({})
