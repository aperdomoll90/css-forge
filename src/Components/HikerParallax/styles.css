#hiker-parallax-wrapper{
  position: relative;
  width: 100vw;
  min-height: 200vh;
  background: var(--mainBackgroundColor);
}

.hiker-parallax {
    position: relative;
    width: 100%;
    height: 50%;
    overflow: hidden;
  }

  .contentOnLoadAnimation{
    opacity: 0;
    animation: onLoadContent var(--coverAnimationTime);
    animation-delay: var(--standardAnimationTime);
    animation-fill-mode: forwards;
  }

  .hiker {
    height:  55vh;
    position: absolute;
    bottom: -80px;
    left: -100px;
    z-index: 10;
  }
  
  .plane2 {
    width: 200vw;
    position: absolute;
    bottom: -45vh;
    right: -15vw;
    z-index: 8;
  }
  
  .plane3 {
    width: 200vw;
    position: absolute;
    bottom: 15vh;
    left: 0;
    z-index: 7;
  }
  
  .backgroundMountain {
    width: 250vw;
    position: absolute;
    top: 5rem;
    left: -20vw;
    z-index: 5;
  }
  
  .sky {
    height: 150vh;
    position: absolute;
    top: calc(50% - 60vh);
    right: calc(50% - 100vw);
    z-index: 4;
  }
  
/* // Large devices (desktops, 992px and up)  */
@media (min-width: 992px) {
    .hiker {
        height:  65vh;
        bottom: -15%;
        left: 0;
      }
      .plane2 {
        width: 100vw;
        bottom: -115%;
        right: 0;
      }

      .plane3 {
        width: 100vw;
        bottom: -25%;
      }

      .backgroundMountain {
        width: 100vw;
        top: 3rem;
        left: -10vw;
      }

      .sky {
        height: auto;
        width: 100vw;
        top: calc(50% - 85vh);
        right: 0;
      }
  }

  @keyframes onLoadContent {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}