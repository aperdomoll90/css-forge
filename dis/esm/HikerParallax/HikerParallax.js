var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useEffect } from 'react';
import './styles.css';
import '../utils/GlobalStyles.css';
// import hiker from './media/hiker.png'
// import plane2 from './media/plane2.png'
// import plane3 from './media/plane3.png'
// import backgroundMountain from './media/plane4.png'
import sky from './media/space.jpeg';
export var HikerParallax = function () {
    useEffect(function () {
        window.addEventListener('scroll', function () {
            var scroll = window.pageYOffset;
            var screenWidth = window.screen.width;
            var translate = document.querySelectorAll('.translate');
            var backgroundMountain = document.querySelector('.backgroundMountain');
            var shadow = document.querySelector('.heroShadow');
            // >>>>>>>>>>>>>>>>>>> TRANSLATE Y
            translate.forEach(function (element) {
                var speed = element.dataset.speed;
                element.style.transform = "translateY(".concat(scroll * speed, "px)");
            });
            // <<<<<<<<<<<<<<<<<<<< HEIGHT
            if (shadow) {
                shadow.style.height = "".concat(scroll * 0.5 + 270, "px");
            }
            // <<<<<<<<<<<<<<<<<<<< WIDTH
            if (backgroundMountain) {
                if (screenWidth < 992) {
                    backgroundMountain.style.width = "".concat(-scroll / 25 + 250, "vw");
                }
                if (screenWidth > 991) {
                    backgroundMountain.style.width = "".concat(-scroll / 25 + 100, "vw");
                }
            }
        });
    }, []);
    return (_jsxs("div", __assign({ id: 'hiker-parallax' }, { children: [_jsx("p", { children: "test" }), _jsx("img", { src: sky, className: 'sky translate', "data-speed": '-0.3', alt: 'sky' })] })));
};
//# sourceMappingURL=HikerParallax.js.map