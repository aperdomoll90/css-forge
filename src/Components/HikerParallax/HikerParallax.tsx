import { useEffect } from 'react'
import './styles.css'
import hiker from './media/hiker.png'
import plane2 from './media/plane2.png'
import plane3 from './media/plane3.png'
import backgroundMountain from './media/plane4.png'
import sky from './media/space.jpeg'

function HikerParallax() {
  useEffect(() => {
    window.addEventListener('scroll', () => {
      let scroll = window.pageYOffset
      let screenWidth = window.screen.width

      const translate = document.querySelectorAll('.translate')
      const backgroundMountain = document.querySelector('.backgroundMountain')
      const shadow = document.querySelector('.heroShadow')

      // >>>>>>>>>>>>>>>>>>> TRANSLATE Y
      translate.forEach(element => {
        let speed = element.dataset.speed
        element.style.transform = `translateY(${scroll * speed}px)`
      })

      // <<<<<<<<<<<<<<<<<<<< HEIGHT
      if (shadow) {
        shadow.style.height = `${scroll * 0.5 + 270}px`
      }

      // <<<<<<<<<<<<<<<<<<<< WIDTH
      if (backgroundMountain) {
        if (screenWidth < 992) {
          backgroundMountain.style.width = `${-scroll / 25 + 250}vw`
        }
        if (screenWidth > 991) {
          backgroundMountain.style.width = `${-scroll / 25 + 100}vw`
        }
      }
    })
  }, [])
  return (
    <section id='hiker-parallax-wrapper'>
      <div className='hiker-parallax'>
        <img src={hiker} className='hiker translate contentOnLoadAnimation' data-speed='0.3' alt='hiker' />
        <img src={plane2} className='plane2 translate contentOnLoadAnimation' data-speed='-0.7' alt='mountain forest' />
        <img src={plane3} className='plane3 translate contentOnLoadAnimation' data-speed='-0.4' alt='medium mountain' />
        <img src={backgroundMountain} className='backgroundMountain contentOnLoadAnimation' alt='High mountain peeks covered in snow' />
        <img src={sky} className='sky translate contentOnLoadAnimation' data-speed='-0.3' alt='sky' />
      </div>
    </section>
  )
}

export default HikerParallax
