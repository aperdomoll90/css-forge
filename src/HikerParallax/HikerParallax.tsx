import { useEffect } from 'react'
import './styles.css'
import '../utils/GlobalStyles.css'
import hiker from './media/hiker.png'
import plane2 from './media/plane2.png'
import plane3 from './media/plane3.png'
import backgroundMountain from './media/plane4.png'
import sky from './media/space.jpeg'

export const HikerParallax: React.FC<{}> = () => {
  useEffect(() => {
    window.addEventListener('scroll', () => {
      let scroll = window.pageYOffset
      let screenWidth = window.screen.width

      const translate = document.querySelectorAll('.translate') as any
      const backgroundMountain = document.querySelector('.backgroundMountain') as any
      const shadow = document.querySelector('.heroShadow') as any

      // >>>>>>>>>>>>>>>>>>> TRANSLATE Y
      translate.forEach((element: any) => {
        let speed = element.dataset.speed
        element.style.transform = `translateY(${scroll * speed}px)`
      })

      // <<<<<<<<<<<<<<<<<<<< HEIGHT
      if (shadow) {
        shadow.style.height = `${scroll * 0.5 + 270}px`
      }

      // <<<<<<<<<<<<<<<<<<<< WIDTH
      if (backgroundMountain) {
        if (screenWidth < 992) {
          backgroundMountain.style.width = `${-scroll / 25 + 250}vw`
        }
        if (screenWidth > 991) {
          backgroundMountain.style.width = `${-scroll / 25 + 100}vw`
        }
      }
    })
  }, [])
  return (
    <div id='hiker-parallax'>
      <p>test</p>
      {/* <img src={hiker} className='hiker translate' data-speed='0.3' alt='hiker' />
      <img src={plane2} className='plane2 translate' data-speed='-0.7' alt='mountain forest' />
      <img src={plane3} className='plane3 translate' data-speed='-0.4' alt='medium mountain' />
      <img src={backgroundMountain} className='backgroundMountain' alt='High mountain peeks covered in snow' /> */}
      <img src={sky} className='sky translate' data-speed='-0.3' alt='sky' />
    </div>
  )
}
