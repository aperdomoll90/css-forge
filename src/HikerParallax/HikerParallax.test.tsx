import { render } from '@testing-library/react'
import {HikerParallax} from './HikerParallax'

describe('HikerParallax component', () => {
  // const { axe, toHaveNoViolations } = require('jest-axe')
  // expect.extend(toHaveNoViolations)

  const renderComponent = () => render(<HikerParallax />)
  it('should match snapshots', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
  //   it('should demonstrate ADA compliance on this component', async () => {
  //     const { container } = renderComponent()
  //     const results = await axe(container)
  //     expect(container).toHaveNoViolations()
  //   })
})
