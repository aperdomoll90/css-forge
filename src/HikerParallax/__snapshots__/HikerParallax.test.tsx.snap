// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HikerParallax component should match snapshots 1`] = `
<div>
  <div
    id="hiker-parallax"
  >
    <p>
      test
    </p>
    <img
      alt="sky"
      class="sky translate"
      data-speed="-0.3"
      src="space.jpeg"
    />
  </div>
</div>
`;
